### ISSUE TITLE
Error on apply rule 'NotEmpty' for key 'File'

### ISSUE THREAD

**Issue by SirBrecilos** [micro-elements/MicroElements.Swashbuckle.FluentValidation#24]
Error when generating Swagger documentation. Results in wrong and missing tags in request body.

error: 

> warn:` MicroElements.Swashbuckle.FluentValidation.FluentValidationRules[0]
>       Error on apply rule 'NotEmpty' for key 'File'.
> System.NullReferenceException: Object reference not set to an instance of an object.
>    at MicroElements.Swashbuckle.FluentValidation.FluentValidationRules.<>c.<CreateDefaultRules>b__6_3(RuleContext context) in /home/travis/build/micro-elements/MicroElements.Swashbuckle.FluentValidation/src/MicroElements.Swashbuckle.FluentValidation/FluentValidationRules.cs:line 141
>    at MicroElements.Swashbuckle.FluentValidation.FluentValidationOperationFilter.ApplyInternal(Operation operation, OperationFilterContext context) in /home/travis/build/micro-elements/MicroElements.Swashbuckle.FluentValidation/src/MicroElements.Swashbuckle.FluentValidation/FluentValidationOperationFilter.cs:line 91
> warn: MicroElements.Swashbuckle.FluentValidation.FluentValidationRules[0]
>       Error on apply rule 'NotEmpty' for key 'File'.
> System.NullReferenceException: Object reference not set to an instance of an object.
>    at MicroElements.Swashbuckle.FluentValidation.FluentValidationRules.<>c.<CreateDefaultRules>b__6_3(RuleContext context) in /home/travis/build/micro-elements/MicroElements.Swashbuckle.FluentValidation/src/MicroElements.Swashbuckle.FluentValidation/FluentValidationRules.cs:line 141
>    at MicroElements.Swashbuckle.FluentValidation.FluentValidationOperationFilter.ApplyInternal(Operation operation, OperationFilterContext context) in /home/travis/build/micro-elements/MicroElements.Swashbuckle.FluentValidation/src/MicroElements.Swashbuckle.FluentValidation/FluentValidationOperationFilter.cs:line 91
> warn: MicroElements.Swashbuckle.FluentValidation.FluentValidationRules[0]
>       Error on apply rule 'Length' for key 'Name'.
> System.NullReferenceException: Object reference not set to an instance of an object.
>    at MicroElements.Swashbuckle.FluentValidation.FluentValidationRules.<>c.<CreateDefaultRules>b__6_5(RuleContext context) in /home/travis/build/micro-elements/MicroElements.Swashbuckle.FluentValidation/src/MicroElements.Swashbuckle.FluentValidation/FluentValidationRules.cs:line 150
>    at MicroElements.Swashbuckle.FluentValidation.FluentValidationOperationFilter.ApplyInternal(Operation operation, OperationFilterContext context) in /home/travis/build/micro-elements/MicroElements.Swashbuckle.FluentValidation/src/MicroElements.Swashbuckle.FluentValidation/FluentValidationOperationFilter.cs:line 91
> warn: MicroElements.Swashbuckle.FluentValidation.FluentValidationRules[0]
>       Error on apply rule 'Length' for key 'Name'.
> System.NullReferenceException: Object reference not set to an instance of an object.
>    at MicroElements.Swashbuckle.FluentValidation.FluentValidationRules.<>c.<CreateDefaultRules>b__6_5(RuleContext context) in /home/travis/build/micro-elements/MicroElements.Swashbuckle.FluentValidation/src/MicroElements.Swashbuckle.FluentValidation/FluentValidationRules.cs:line 150
>    at MicroElements.Swashbuckle.FluentValidation.FluentValidationOperationFilter.ApplyInternal(Operation operation, OperationFilterContext context) in /home/travis/build/micro-elements/MicroElements.Swashbuckle.FluentValidation/src/MicroElements.Swashbuckle.FluentValidation/FluentValidationOperationFilter.cs:line 91
> warn: MicroElements.Swashbuckle.FluentValidation.FluentValidationRules[0]
>       Error on apply rule 'NotEmpty' for key 'File'.
> System.NullReferenceException: Object reference not set to an instance of an object.
>    at MicroElements.Swashbuckle.FluentValidation.FluentValidationRules.<>c.<CreateDefaultRules>b__6_3(RuleContext context) in /home/travis/build/micro-elements/MicroElements.Swashbuckle.FluentValidation/src/MicroElements.Swashbuckle.FluentValidation/FluentValidationRules.cs:line 141
>    at MicroElements.Swashbuckle.FluentValidation.FluentValidationOperationFilter.ApplyInternal(Operation operation, OperationFilterContext context) in /home/travis/build/micro-elements/MicroElements.Swashbuckle.FluentValidation/src/MicroElements.Swashbuckle.FluentValidation/FluentValidationOperationFilter.cs:line 91
> warn: MicroElements.Swashbuckle.FluentValidation.FluentValidationRules[0]
>       Error on apply rule 'NotEmpty' for key 'File'.
> System.NullReferenceException: Object reference not set to an instance of an object.
>    at MicroElements.Swashbuckle.FluentValidation.FluentValidationRules.<>c.<CreateDefaultRules>b__6_3(RuleContext context) in /home/travis/build/micro-elements/MicroElements.Swashbuckle.FluentValidation/src/MicroElements.Swashbuckle.FluentValidation/FluentValidationRules.cs:line 141
>    at MicroElements.Swashbuckle.FluentValidation.FluentValidationOperationFilter.ApplyInternal(Operation operation, OperationFilterContext context) in /home/travis/build/micro-elements/MicroElements.Swashbuckle.FluentValidation/src/MicroElements.Swashbuckle.FluentValidation/FluentValidationOperationFilter.cs:line 91
> warn: MicroElements.Swashbuckle.FluentValidation.FluentValidationRules[0]
>       Error on apply rule 'Length' for key 'Name'.
> System.NullReferenceException: Object reference not set to an instance of an object.
>    at MicroElements.Swashbuckle.FluentValidation.FluentValidationRules.<>c.<CreateDefaultRules>b__6_5(RuleContext context) in /home/travis/build/micro-elements/MicroElements.Swashbuckle.FluentValidation/src/MicroElements.Swashbuckle.FluentValidation/FluentValidationRules.cs:line 150
>    at MicroElements.Swashbuckle.FluentValidation.FluentValidationOperationFilter.ApplyInternal(Operation operation, OperationFilterContext context) in /home/travis/build/micro-elements/MicroElements.Swashbuckle.FluentValidation/src/MicroElements.Swashbuckle.FluentValidation/FluentValidationOperationFilter.cs:line 91
> warn: MicroElements.Swashbuckle.FluentValidation.FluentValidationRules[0]
>       Error on apply rule 'Length' for key 'Name'.
> System.NullReferenceException: Object reference not set to an instance of an object.
>    at MicroElements.Swashbuckle.FluentValidation.FluentValidationRules.<>c.<CreateDefaultRules>b__6_5(RuleContext context) in /home/travis/build/micro-elements/MicroElements.Swashbuckle.FluentValidation/src/MicroElements.Swashbuckle.FluentValidation/FluentValidationRules.cs:line 150
>    at MicroElements.Swashbuckle.FluentValidation.FluentValidationOperationFilter.ApplyInternal(Operation operation, OperationFilterContext context) in /home/travis/build/micro-elements/MicroElements.Swashbuckle.FluentValidation/src/MicroElements.Swashbuckle.FluentValidation/FluentValidationOperationFilter.cs:line 91

My request object:
``` csharp 
public class CreateImageRequest
    {
        public IFormFile File { get; set; }
        public string Name { get; set; }
    }
```

validator:
``` csharp
public class CreateImageRequestValidator : AbstractValidator<CreateImageRequest>
    {
        public CreateImageRequestValidator()
        {
            RuleFor(_ => _.File).NotEmpty().WithErrorCode(ErrorCode.MustNotBeEmpty.ToString());
            RuleFor(_ => _.File).Must((request, ct) =>
            {
                if ((request?.File?.ContentType == @"image/jpeg") || (request?.File?.ContentType == @"image/png"))
                {
                    return true;
                }
                return false;
            }).WithMessage("Invalid content type").WithErrorCode(ErrorCode.ExpectationNotMet.ToString());
            RuleFor(_ => _.Name).Length(0, 250).WithErrorCode(ErrorCode.IncorrectLength.ToString());
        }
    }
```

Controller method;
``` csharp
public async Task<IActionResult> CreateProfileImage([FromForm]CreateImageRequest request)
        {
             ...
        }
```

Parameters in swagger docs: (somehow it's labeled as query parameter and no required tags..)

![image](https://user-images.githubusercontent.com/14992892/48950952-2c4baa80-ef3d-11e8-99d6-603c655dfd47.png)

_(edited: added validator class)_

**Comment by petriashev** [comment]
What version of package do you use? What version of swagger?

**Comment by SirBrecilos** [comment]
Currently using:
``` csharp
<ItemGroup>
    <PackageReference Include="Autofac.Extensions.DependencyInjection" Version="4.3.0" />
    <PackageReference Include="FluentValidation" Version="8.0.100" />
    <PackageReference Include="FluentValidation.AspNetCore" Version="8.0.100" />
    <PackageReference Include="MicroElements.Swashbuckle.FluentValidation" Version="1.1.0" />
    <PackageReference Include="Swashbuckle.AspNetCore" Version="3.0.0" />
    <PackageReference Include="Swashbuckle.AspNetCore.ReDoc" Version="3.0.0" />
     ...
    <PackageReference Include="Microsoft.ApplicationInsights.AspNetCore" Version="2.3.0" />
    <PackageReference Include="Microsoft.AspNetCore.App" />
  </ItemGroup>
```

**Comment by vlad294** [comment]
Also not working for [FromQuery] Request: `public Task Get([FromQuery] SampleRequest request)`
```
   
    public class SampleRequest
    {
        public int Offset { get; set; }

        public int Count { get; set; }
    }
```

and for [FromForm] `public Task Publish([FromForm] SampleRequest request)`
```
    public class SampleRequest 
    {
        public DateTime OpenDateTime { get; set; }

        public DateTime CloseDateTime { get; set; }
    }
```

Warning in log:
```
System.NullReferenceException: Object reference not set to an instance of an object.
   at MicroElements.Swashbuckle.FluentValidation.FluentValidationRules.<>c.<CreateDefaultRules>b__6_9(RuleContext context) in /home/travis/build/micro-elements/MicroElements.Swashbuckle.FluentValidation/src/MicroElements.Swashbuckle.FluentValidation/FluentValidationRules.cs:line 172
   at MicroElements.Swashbuckle.FluentValidation.FluentValidationOperationFilter.ApplyInternal(Operation operation, OperationFilterContext context) in /home/travis/build/micro-elements/MicroElements.Swashbuckle.FluentValidation/src/MicroElements.Swashbuckle.FluentValidation/FluentValidationOperationFilter.cs:line 91
```

**Comment by petriashev** [comment]
I cant reproduce it.
See branch https://github.com/micro-elements/MicroElements.Swashbuckle.FluentValidation/tree/bugfix/Issue24

And file:
 https://github.com/micro-elements/MicroElements.Swashbuckle.FluentValidation/blob/bugfix/Issue24/samples/SampleWebApi/GithubIssues/Issue24.cs

Can you provide code that reproduce error (zip project or fork with sample file)?

**Comment by vlad294** [comment]
Sorry for not replying sooner. 
Ok, I will try to make sample project :-)

**Comment by petriashev** [comment]
Any changes?

**Comment by JdDCalixL** [comment]
Hi, almost same problem here.
Error on apply rule 'NotEmpty' for key 'Nombre'.
System.NullReferenceException: Object reference not set to an instance of an object.
   at MicroElements.Swashbuckle.FluentValidation.FluentValidationRules.<>c.<CreateDefaultRules>b__6_3(RuleContext context) in /home/travis/build/micro-elements/MicroElements.Swashbuckle.FluentValidation/src/MicroElements.Swashbuckle.FluentValidation/FluentValidationRules.cs:line

for Post[FromBody] the model json does show the ValidationMetadata
but for Post[FromQuery] doesnt.

<TargetFramework>netcoreapp2.1</TargetFramework>
<PackageReference Include="Microsoft.AspNetCore.App" />
<PackageReference Include="MicroElements.Swashbuckle.FluentValidation" Version="2.0.0-beta.2" />
<PackageReference Include="Swashbuckle.AspNetCore" Version="4.0.1" />

any help will be appreciated. thanks

**Comment by petriashev** [comment]
@JdDCalixL Can you attach sample code?

**Comment by petriashev** [comment]
closed because no activity

**Comment by herrirya** [comment]
Hello, I believe I have reproduced this issue, if not one that is very similar. Here are steps to reproduce:

1. I downloaded the master branch, and modified SampleWebApi->BasicRequestValidator.cs, adding just a simple new MaximumLength validation as a new first line to BasicRequestValidator() method: 

`RuleFor(o => o.ValueFromHeader).MaximumLength(100);`

2. Ran the SampleWebApi project, navigating to Swagger json endpoint like:
http://localhost:[PORT]/swagger/v1/swagger.json

3. The following Exception is thrown:

**Error on apply rule 'Length' for key 'ValueFromHeader'**

```
Exception thrown: 'System.NullReferenceException' in MicroElements.Swashbuckle.FluentValidation.dll
MicroElements.Swashbuckle.FluentValidation.FluentValidationRules:Warning: Error on apply rule 'Length' for key 'ValueFromHeader'.

System.NullReferenceException: Object reference not set to an instance of an object.
   at MicroElements.Swashbuckle.FluentValidation.FluentValidationRules.<>c.<CreateDefaultRules>b__7_5(RuleContext context) in C:\Users\herrirya\Downloads\MicroElements.Swashbuckle.FluentValidation-master\src\MicroElements.Swashbuckle.FluentValidation\FluentValidationRules.cs:line 162
   at MicroElements.Swashbuckle.FluentValidation.FluentValidationOperationFilter.ApplyInternal(Operation operation, OperationFilterContext context) in C:\Users\herrirya\Downloads\MicroElements.Swashbuckle.FluentValidation-master\src\MicroElements.Swashbuckle.FluentValidation\FluentValidationOperationFilter.cs:line 90
```

Thank you for your work on this, it seems like a useful project.

**Comment by petriashev** [comment]
Thank you. Will see soon.

**Comment by petriashev** [comment]
Problem reproduced. I have a fix. Will release after some tests.

**Comment by petriashev** [comment]
closed by #30
